File: 2015/10/rust/src/main.rs
// --- Constants ---
const INPUT_FILE:&'static str = "../input/input.txt";

fn look_and_say(s: String) -> String {
    let a: Vec<char> = s.chars().collect();
    let mut res: Vec<char> = Vec::new();
    let mut last = a[0];
    let mut count = 1;
    for i in 1 .. a.len() {
        if a[i] == last {
            count += 1;
        } else {
            res.extend(count.to_string().chars());
            res.push(last);
            count = 1;
        }
        last = a[i];
    }
    res.extend(count.to_string().chars());
    res.push(last);
    res.into_iter().collect()
}

// --- Main function ---
fn main() -> Result<(), Box<dyn std::error::Error>> {
    // --- Read and parse the input file ---
    let mut text = std::fs::read_to_string(INPUT_FILE)?.trim().to_string();

    // --- Puzzle 1 ---
    for _ in 0 .. 40 {
        text = look_and_say(text);
    }
    println!("1. Length of the result: {}", text.len());

    // --- Puzzle 2 ---
    for _ in 0 .. 10 {
        text = look_and_say(text);
    }
    println!("2. Length of the result: {}", text.len());

    Ok(())
}

File: 2015/10/python/main.py
# --- Constants --
INPUT_FILE = "../input/input.txt"

# --- Variables ---
def look_and_say(s):
    ls = [*s]
    res = []
    last = ls[0]
    count = 1
    for i in range(1, len(ls)):
        if ls[i] == last: count += 1
        else:
            res.extend([*str(count)])
            res.append(last)
            count = 1
        last = ls[i]
    res.extend([*str(count)])
    res.append(last)
    return "".join(res)

# --- Read and Parse the input file --
file = open(INPUT_FILE, "r")
text = file.read().strip()

# --- Puzzle 1 ---
for _ in range(0, 40): text = look_and_say(text)
print(f"1. Length of the result: {len(text):,d}")

# --- Puzzle 2 ---
for _ in range(0, 10): text = look_and_say(text)
print(f"2. Length of the result: {len(text):,d}")
